db_test.go

```diff
@@ -1011,3 +1013,109 @@ func TestOverlappingBlocksDetectsAllOverlaps(t *testing.T) {
 		{Min: 8, Max: 9}: {nc1[8], nc1[9]},                                 // 7-10, 8-9
 	}, OverlappingBlocks(nc1))
 }
+
+func TestInitializeHeadTimestamp(t *testing.T) {
```

Comment:
shouldn't this test be in  `head_test.go`?

Reply:
actually there is already a head test where you can just add few lines similar to what I did in the other PR
https://github.com/prometheus/tsdb/pull/334/files#diff-0dc7b53c06c14aba584e5b4d6986fa07R96

Reply:
The test cover cases with existing blocks, which is beyond the scope of `Head`. `DB` has to startup correctly and make some calls of `Head`'s methods in the right order and with the right arguments. That's nothing we can test with an isolated `Head`.